pkg/controller/core/clusterqueue_controller.go

```diff
@@ -497,3 +500,18 @@ func (r *ClusterQueueReconciler) updateCqStatusIfChanged(
 	}
 	return nil
 }
+
+func (r *ClusterQueueReconciler) getWorkloadsStatus(cq *kueue.ClusterQueue) *kueue.ClusterQueuePendingWorkloadsStatus {
+	pendingWorkloadsStatus := cq.Status.DeepCopy().PendingWorkloadsStatus
+	if pendingWorkloadsStatus == nil {
+		return &kueue.ClusterQueuePendingWorkloadsStatus{
+			LastChangeTime: metav1.Time{Time: time.Now()},
+		}
+	}
+	pendingWorkloads := r.qManager.GetSnapshot(cq.Name)
+	if !equality.Semantic.DeepEqual(pendingWorkloadsStatus.Head, pendingWorkloads) {
```

Comment:
Check this relatively heavy condition only if `now - lastChangeTime >= UpdateStatusIntervalSeconds`.

Reply:
added condition here: 
https://github.com/kubernetes-sigs/kueue/pull/1069/files#diff-54fb2e1e38c1d823a077207b574c6dc9f21d0b354584becd6940b26282aec41fR549